pontoon/contributors/views.py

```diff
@@ -310,27 +311,38 @@ def settings(request):
     all_locales = list(Locale.objects.all())
     all_locales.insert(0, default_homepage_locale)
 
-    # Set custom homepage selector value
-    custom_homepage_locale = profile.custom_homepage
-    if custom_homepage_locale:
-        custom_homepage_locale = Locale.objects.filter(
-            code=custom_homepage_locale
-        ).first()
-    else:
+    # Set default for custom homepage locale based on code
+    try:
+        if profile.custom_homepage:
+            custom_homepage_locale = get_locale_or_redirect(profile.custom_homepage)
+        else:
+            custom_homepage_locale = default_homepage_locale
```

Comment:
You can move this line in front of the `try - except` block and then you can remove it from line 321. 

At that point you no longer need the `else` block and you can move the `try - except` block within the `if` block.